src/utils/formatting.py

```
@@ -0,0 +1,258 @@
+        Dictionary with new observations by level
+    """
+    new_observations_by_level: dict[str, Any] = {}
+
+    # Helper function to get observations from either ReasoningResponse or dict
+    def get_observations(
+        context: ReasoningResponse | dict[str, Any], level: str
+    ) -> list[Any]:
+        if isinstance(context, ReasoningResponse):
+            # It's a ReasoningResponse object
+            return getattr(context, level, [])
+        else:
```

Ruff (0.12.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)